wremnants/datasets/datagroups.py

```diff
@@ -536,7 +536,8 @@ def setGenAxes(self, gen_axes=None):
         else:
             # infer gen axes from metadata
             args = self.getMetaInfo()["args"]
-            if args.get("unfolding", False) is False and args.get("addHelicityHistos", False) is False:
+            ## TODO: this should be addHelicityHisto->theoryAgnostic since the option changed name, BUT this breaks the poiAsNoi case
+            if args.get("unfolding", False) is False and (args.get("theoryAgnostic", False) is False or args.get("poiAsNoi", False) is True):
```

Comment:
Can't this just be something like if not any(args.get(x, False) for x in ["unfolding", "theoryAgnostic", ...])

Reply:
Maybe, it has to be carefully tuned. I'll see what I can do. The logic is that if you don't do unfolding or theory agnostic you don't have to set those gen axes, because this will trigger integrating them out when doing other things for fakes. But the POIasNOI version of the theory agnostic is an exception, because in that case the axes are not part of the signal axes, they are systematic axes

Reply:
Could do:
```
if not args.get("unfolding", False) and (not args.get("theoryAgnostic", False) or args.get("poiAsNoi", False))
```

Reply:
ah ok, sure